desktop/core/src/desktop/js/apps/storageBrowser/StorageDirectoryPage/StorageDirectoryPage.tsx

```diff
@@ -186,9 +186,11 @@ const StorageDirectoryPage = ({
   const onRowClicked = (record: StorageDirectoryTableData) => {
     return {
       onClick: () => {
-        onFilePathChange(record.path);
-        if (record.type === 'dir') {
-          setPageNumber(1);
+        if (selectedFiles.length === 0) {
```

Comment:
if (selectedFiles.length === 0 && record.type === 'dir') { 

nit: single if instead of nested